hazelcast/src/main/java/com/hazelcast/config/EndpointConfig.java

```diff
@@ -274,10 +278,35 @@ public EndpointConfig setSSLConfig(SSLConfig sslConfig) {
         return this;
     }
 
+    /**
+     * Gets the Alto socket config.
+     *
+     * @return the Alto socket config
+     * @see com.hazelcast.config.alto.AltoConfig
+     * @since 5.3
+     */
+    @Beta
+    @Nonnull
     public AltoSocketConfig getAltoSocketConfig() {
         return altoSocketConfig;
     }
 
+    /**
+     * Sets the Alto socket config.
+     *
+     * @param altoSocketConfig Alto socket config to set
+     * @return this endpoint config
+     * @throws IllegalArgumentException if altoSocketConfig is null
```

Comment:
Should this not be a null pointer exception? Doesn't really matter, but just curious.

Reply:
https://github.com/hazelcast/hazelcast/pull/23319/commits/2d64546dd313058fb343aabc5f61955f03baf460